### ISSUE TITLE
Add Swedish Volunteers

### ISSUE THREAD

**Issue by jbrubake** [vasl-developers/vasl#1038]
I updated the SV extension to include the new Ski counters and Ahkios and it's now on [VASL.info](http://vasl.info)

However the counters are still the old blue in white. I'm fairly experienced with Gimp so I could update the counters but I don't know how to create the various overlays (HS, SMC, etc) that make up the counters. 

What is the proper technique for getting digital versions of the counter images?

**Comment by derimmer** [comment]
Well there are probably a couple of different ways. Given that it is an extension, what I would do is the following:

1. Change the file suffix to .zip. 
2. Open the zip file, extract the images that need to be changed. I believe you will find an "images" folder within the .zip file.
3. Change the images
4. Copy them back to the .zip file. 
5. Rename the .zip file back to its original name. 

The management of the counters (sq/hs) is handled by VASSAL. To change  a counter, open VASSAL, select a module and rightclick. Select Edit Module. You can then expend the module tree and find the counters in VASL Counters. 

For an extension, you first have to add the extension to the module by using the rightclick option, then edit the extension as above. 

If you are updating images only and not changing the behaviour of the counters then just replacing the images in the extension shoudl be sufficient so long as you keep the image file names the same. You should not have to edit the extension in VASSAL. 

Happy to help with any issues that arise. Once it is all good to go, I can then move the counters from the extension into the core VASL module.

**Comment by jbrubake** [comment]
After rereading my initial comment, I realize I was not very clear on what my issue is.

I know how to add/remove images from the extension. My problem is figuring out how to change the colors of the current counters. I can easily change the background from white to Swedish blue, but I don't know how to change the light blue images and text to the correct black/Swedish blue color gradient. That kind of color manipulation is beyond my experience.

I figured out how to shade the images and text to black, but that doesn't math the other nationalities counters which have portions that have shades of the underlying counter color.

Hopefully this made more sense this time. I basically need someone with more knowledge of graphic editing to either fix the counters or explain to me how to do so, otherwise the counters won't be the official colors.

**Comment by derimmer** [comment]
@BigAl737 any advice on this one Al?

**Comment by BigAl737** [comment]
Sure:

Open up a gif from the SV extension in GIMP
![image](https://user-images.githubusercontent.com/10384413/149062933-c0369049-3e14-4fb2-8a96-5654f45b1c9a.png)

Change the Image Mode to RGB: Image, Mode, RGB
![image](https://user-images.githubusercontent.com/10384413/149061918-ffe77878-c40a-4052-adfb-eb83e1d252ee.png)

Desaturate: Colors, Desaturate (I like the Lightness option for this application)
![image](https://user-images.githubusercontent.com/10384413/149062007-c2a8a5b6-5f8f-4937-9f5b-7b4eb92cdc14.png)
![image](https://user-images.githubusercontent.com/10384413/149062149-29df4574-39af-419a-939a-891664eb9bf3.png)

Darken the image: Colors, Levels
![image](https://user-images.githubusercontent.com/10384413/149062208-138be379-8042-4a6a-b9d3-10005ac90741.png)

Darken to taste but a Clamp input around .30 seems to work nicely
![image](https://user-images.githubusercontent.com/10384413/149062285-64db1dc6-6341-4b94-b91a-c180cc3aaf38.png)
![image](https://user-images.githubusercontent.com/10384413/149062307-ca888759-f03f-408b-896e-ee1496b2ca04.png)

Remove the background: Colors, Color to Alpha
![image](https://user-images.githubusercontent.com/10384413/149062392-e3affcf3-a929-420f-9a71-590a23c205c2.png)
![image](https://user-images.githubusercontent.com/10384413/149063088-f93fddbf-cc44-445a-9aee-3dcaa1b546a8.png)
![image](https://user-images.githubusercontent.com/10384413/149062454-833e4f7d-fa30-46ad-81d8-2d5690760359.png)
(If this didn't work, make sure the image has an alpha layer)

Paste a new background layer from one of these:
![svblank58](https://user-images.githubusercontent.com/10384413/149062550-6cb050af-c310-4f80-9b7e-ed63024e865a.png)
![svblank](https://user-images.githubusercontent.com/10384413/149062556-0dd522b1-4eee-4c7d-b513-d785237d0b3f.png)
![image](https://user-images.githubusercontent.com/10384413/149062585-62ada87b-25f6-4e6d-ae38-b5b52895ea57.png)

Export as a .png: (Shift+Ctrl+E)

The official Swedish VASL RGB triplet is: 105,168,217 (HTML 69a89d)
FYI, I've never gotten original source files for counters. I've always had to scan or convert them.

Allan

**Comment by BigAl737** [comment]
BTW, when I click on the VASL.org download link for the SV.mdx extension, I get unreadable computer code, not a downloaded file.

**Comment by derimmer** [comment]
> BTW, when I click on the VASL.org download link for the SV.mdx extension, I get unreadable computer code, not a downloaded file.

I believe that if you right click on the screen with the junk code on it, select save as then save the module locally with the SV.mdx name, it will all work out. 

This is not the first download I have seen work this way, I don't know why.

**Comment by BigAl737** [comment]
You're right, it does. Thanks Doug. Maybe cause it's named a .mdx and not a .vmdx? I don't need an answer but if it's worth opening as another issue, please do so.

**Comment by derimmer** [comment]
Not worth an issue because this happens outside of a vasl/vassal context. I have had it happen when downloading from other sites.

**Comment by jbrubake** [comment]
@BigAl737 Thanks. I should be able to convert those fairly quickly (once I get time of course). I realized I also left the old style ski and Ahkio counters in the module so I'll remove them as well.

Once it's complete, what's the process for getting the Swedes into the main VASL code?

**Comment by derimmer** [comment]
> Once it's complete, what's the process for getting the Swedes into the main VASL code?

With Big Al's help, I will cut-and-paste (1) the new images and (2) the buildFile references for the counters from the extension to the module. 

This is what we did for KFW. It ended in tears but we made it work. This is not so much a process as a quick-and-dirty way of bringing them in. 

Once that is done, they will be in the VASL module's next official release. 6.6.4 due out in April.

**Comment by derimmer** [comment]
> 
> I sent it up to have it posted on https://vasl.info but I can also send it direct if you want it

I thought that the intent was to have it added to the core VASL module and not remain and extension? I am fine either way.

**Comment by jbrubake** [comment]
That was my understanding as well but beyond my knowledge. I figured it should at least be an extension until it's added to the core module.

**Comment by jbrubake** [comment]
Sam Tyson. I can send it to you direct if you want

**Comment by jbrubake** [comment]
Sent directly

**Comment by derimmer** [comment]
I have add the sv extension to core VASL. 

Here is a dropbox link to a beta module for testing. Jeremy, I would particularly appreciate your attention to this to make sure that I have put everything in its proper place. 
https://www.dropbox.com/s/o5gkwokssm4tpns/vasl-6.6.4-beta5.vmod?dl=0

Also I have added the CVP functionality to all the Swedish units/weapons.

**Comment by jbrubake** [comment]
Everything looks good to me. Some of the counters could use some touch up. I did some and then finished a few weeks later after learning some better techniques. I guess I forgot to go back and redo the ones that didn't come out as well

**Comment by derimmer** [comment]
Good. 

 

IF it is just a case of altering some images, I can reload them in a flash whenever you get them done.
